src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,130 @@
-import React from 'react';
+/* eslint-disable max-len */
+import classNames from 'classnames';
+import React, { useMemo, useState, useEffect } from 'react';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+type Props = {
+  todos: Todo[];
+  selectedUserId: number;
+  selectUser: (id: number) => void;
+};
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+export const TodoList: React.FC<Props> = ({
+  todos,
+  selectedUserId,
+  selectUser,
+}) => {
+  const [query, setQuery] = useState<string>('');
+  const [statusFilter, setStatusFilter] = useState<string>('all');
+  const [random, setRandom] = useState<boolean>(false);
+  const [queryLLC, setQueryLLC] = useState<string>('');
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+  useEffect(() => {
+    setQueryLLC(query.toLocaleLowerCase());
+  }, [query]);
+
+  const visibleTodos = useMemo(() => (
+    todos
+      .filter(todo => todo.title.toLocaleLowerCase().includes(queryLLC))
+      .filter(todo => {
+        switch (statusFilter) {
+          case 'all':
+            return todo;
+
+          case 'active':
+            return todo.completed === false;
+
+          case 'completed':
+            return todo.completed;
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+          default:
+            return 0;
+        }
+      })
+      .sort((t1, t2) => (
+        random
+          ? Math.random() - 0.5
+          : t1.id - t2.id
+      ))
```

Comment:
read about [fisher-yates algorithm](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle) for shuffle array

Reply:
used